ui/uihandler.go

```diff
@@ -951,3 +945,12 @@ func RedirectToHTTPS(port int) {
 	}
 	fmt.Printf("Not reached, https redirector exiting - was on %v\n", s.Addr)
 }
+
+// setHostAndPort takes hostport in the form of hostname:port, ip:port or :port
+// and returns "localhost:port" if hostport is in the form of :port.
+func setHostAndPort(hostport string) string {
+	if strings.HasPrefix(hostport, ":") {
+		return "localhost" + hostport
```

Comment:
it should set the global and return the port instead?

Reply:
got it.